src/Storages/MergeTree/localBackup.cpp

```diff
@@ -100,7 +125,8 @@ void localBackup(
     size_t try_no = 0;
     const size_t max_tries = 10;
 
-    CleanupOnFail cleanup([disk, destination_path]() { disk->removeRecursive(destination_path); });
+    CleanupOnFail cleanup(disk_transaction ? std::function<void()>([]{}) :
+                          [disk, destination_path]() { disk->removeRecursive(destination_path); });
```

Comment:
Seems like it may remove shared blobs from S3 (when zero-copy-replication is being used) resulting in a data loss.

Reply:
(This is a comment for https://github.com/ClickHouse/ClickHouse/pull/30797 , not related to this PR, right? Idk how zero-copy replication works, but sounds plausible.)